File: taobao-user-interface/src/main/java/com/taobao/user/service/AddressService.java
package com.taobao.user.service;

import com.taobao.pojo.TbAddress;
import com.taobao.pojo.TbUser;
import com.taobao.pojo.group.PageResult;

import java.util.List;

/**
 * @Author: dabin0808
 * @Date: 2019/8/27 19:22
 * Explain: 用户地址接口
 */
public interface AddressService {

    /**
     * 返回用户ID全部列表
     *
     * @return
     */
    public List<TbAddress> findUserId(TbUser tbUser);


    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void addAddress(TbAddress address);


    /**
     * 修改
     */
    public void update(TbAddress address);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbAddress findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbAddress address, int pageNum, int pageSize);

    /**
     * 得到用户的所有收货地址
     */
    public List<TbAddress> listByUserId(Long userId);
}

File: taobao-user-web/src/main/webapp/js/service/addressService.js
//地址服务层
app.service('addressService', function ($http) {

    //读取用户ID列表数据绑定到表单中
    this.findUserId = function (entity) {
        return $http.post('../address/findAll.do', entity);
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../address/findPage.do?page=' + page + '&rows=' + rows);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.get('../address/findOne.do?id=' + id);
    }
    //增加
    this.save = function (entity) {
        return $http.post("../address/addAddress.do", entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../address/update.do', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../address/delete.do?ids=' + ids);
    }
    //搜索
    this.search = function (page, rows, searchEntity) {
        return $http.post('../address/search.do?page=' + page + "&rows=" + rows, searchEntity);
    }

    //加载用户地址
    this.findByUserId = function (userId) {
        return $http.get("address/findById.do?userId=" + userId);
    }
});
